### PR TITLE
API-955: Rework and fix migration script

### PR DESCRIPTION
**Pull request by fitn**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: upgrades/schema/Version_4_0_20191031124707_update_from_clients_to_connections.php [hunk 1]
```diff
@@ -30,82 +34,45 @@ public function up(Schema $schema) : void
...
 
         $this->skipIf(empty($clients), 'No API connection to migrate.');
         $this->write(sprintf('%s API connections found. They will be migrate to Connection.', count($clients)));
 
-        $clients = $this->generateConnectionsCode($clients);
+        foreach ($clients as $index => $client) {
+            $connectionCode = preg_replace('/[^A-Za-z0-9\-]/', '_', $client['label']); // slugify code
+            $connectionCode = substr(str_pad($connectionCode, 3, '_'), 0, 97); // 3 to 100 chars length + unique
+            if (isset($this->connections[$connectionCode])) {
+                $connectionCode = sprintf('%s_%s', $connectionCode, rand(0, 99)); // make it unique
```
Comment (willy-ahva) [inline upgrades/schema/Version_4_0_20191031124707_update_from_clients_to_connections.php]: it doesn't make it unique at all

File: upgrades/test_schema/Version_4_0_20191031124707_update_from_clients_to_connections_Integration.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+use PHPUnit\Framework\Assert;
+
+/**
+ * This class will be removed after 4.0 version
+ *
+ * @author    Nicolas Marniesse <nicolas.marniesse@akeneo.com>
+ * @copyright 2019 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class Version_4_0_20191004145507_remove_compute_model_descendant_jobs_Integration extends AbstractPimCatalogTestCase
```
Comment (willy-ahva) [inline upgrades/test_schema/Version_4_0_20191031124707_update_from_clients_to_connections_Integration.php]: why is it in your PR ?

File: upgrades/schema/Version_4_0_20191031124707_update_from_clients_to_connections.php (hunk 1 above)
Comment (pchasle) [reply upgrades/schema/Version_4_0_20191031124707_update_from_clients_to_connections.php]: A simple `do { /* randomize $connectionCode  */ } while ($this->connections[$connectionCode]);` should be enough

Comment (fitn) [reply upgrades/schema/Version_4_0_20191031124707_update_from_clients_to_connections.php]: In fact, it's already not unique with the current script. Anyway I rollback this part of the code.

File: upgrades/test_schema/Version_4_0_20191031124707_update_from_clients_to_connections_Integration.php (hunk 2 above)
Comment (fitn) [reply upgrades/test_schema/Version_4_0_20191031124707_update_from_clients_to_connections_Integration.php]: Don't know but seems not the case anymore

File: upgrades/test_schema/Version_4_0_20191031124707_update_from_clients_to_connections_Integration.php [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+    private function getDbalConnection(): Connection
+    {
+        return $this->get('database_connection');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function addDocuments()
```
Comment (willy-ahva) [inline upgrades/test_schema/Version_4_0_20191031124707_update_from_clients_to_connections_Integration.php]: RM ?

Comment (pierrejolly) [inline upgrades/test_schema/Version_4_0_20191031124707_update_from_clients_to_connections_Integration.php]: To remove ?

Comment (fitn) [reply upgrades/test_schema/Version_4_0_20191031124707_update_from_clients_to_connections_Integration.php]: It's an abstract method. I don't what is it for but I need it ^^

Comment (fitn) [reply upgrades/test_schema/Version_4_0_20191031124707_update_from_clients_to_connections_Integration.php]: It's an abstract method. I don't what is it for but I need it ^^

**Pull request merged**